src/PayloadListener/Services/EventPayloadRecieverService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Monai.Deploy.Messaging;
+using Microsoft.Extensions.Logging;
+using Monai.Deploy.Messaging.Common;
+using Monai.Deploy.Messaging.Messages;
+using Monai.Deploy.WorkflowManager.Logging.Logging;
+using Monai.Deploy.WorkflowManager.PayloadListener.Validators;
+
+namespace Monai.Deploy.WorkflowManager.PayloadListener.Services
+{
+    public class EventPayloadRecieverService : IEventPayloadRecieverService
+    {
+        public EventPayloadRecieverService(
+            ILogger<EventPayloadRecieverService> logger,
+            IEventPayloadValidator payloadValidator,
+            IMessageBrokerSubscriberService messageBrokerSubscriberService)
+        {
+            Logger = logger;
+            PayloadValidator = payloadValidator;
+            _messageSubscriber = messageBrokerSubscriberService;
+        }
+
+        private IEventPayloadValidator PayloadValidator { get; }
+
+        private ILogger<EventPayloadRecieverService> Logger { get; }
+
+        private readonly IMessageBrokerSubscriberService _messageSubscriber;
+
+        public async Task RecieveWorkflowPayload(MessageReceivedEventArgs message)
+        {
+            try
+            {
+                var payload = message.Message.ConvertTo<WorkflowRequestMessage>();
+
+                var validation = PayloadValidator.ValidateWorkflowRequest(payload);
+
+                if (!validation)
+                {
+                    Logger.EventRejectedNoQueue(message.Message.MessageId);
+                    _messageSubscriber.Reject(message.Message);
+
+                    return;
+                }
+
+                //Workflow executor called here
+                _messageSubscriber.Acknowledge(message.Message);
+            }
+            catch (Exception e)
```

Comment:
what happens if the workflow executer fails because the database is inaccessible (for example)? We need to be able to commute that an environmental failure has occurred, so that the message is not acknowledged - but that it is a rejected and re-queued, to be tried again later. At the moment we are simply rejecting it but it will never be tried again

Reply:
I have added basic re-queue logic with a catch all for now, the custom exception logic is part of this ticket. https://github.com/Project-MONAI/monai-deploy-workflow-manager/issues/93